### ISSUE TITLE
Proposal: Cooperative Cookie Syncing

### ISSUE THREAD

**Issue by bretg** [prebid/prebid-server#924]
We've discussed in the Prebid Server committee meetings a feature that allows a Prebid Server host company to pre-sync cookies with more bidders than are in the page. 

1) Prebid Server host company can configure lists of bidder adapters that can be included as part of a cookie-sync response even if that bidder is not specifically requested by the page. 

The lists of bidders should be prioritizable in groups, representing the fact that not all bidders are of equal importance to the publishers of a given host company.

```
cookie-sync:
   coop-sync:
       default: true
       pri[0]: bidderA, bidderB
       pri[1]: bidderC, bidderD
```

Which bidders are chosen within a given group should be random.
If no priority groups are specified, the default is to have all bidders at the same level.

2) The publisher should be able to limit how many sync pixels can happen in a given cookie-sync, and should be able to turn off cooperative syncing.

```
{"bidders": ["appnexus", "rubicon"], coopSync: true, limit: 5}
```

For example, if appnexus and rubicon are on the page and biddersA-D are in the coop-sync config as shown above, PBS would respond with up to 5 sync pixels. e.g. if the user had no IDs in the `uids` cookie, then the response to /cookie-sync would include pixels for appnexus, rubicon, bidderA, bidderB, and (randomly) either bidderC or D.

3) Whether coop syncing is on by default should be configurable by the Prebid Server host company

**Comment by bretg** [comment]
Note: working on this in PBS-Java. 
We tweaked the format of the config above to have it better conform to YAML

**Comment by bretg** [comment]
This is done in PBS-Java. Assigning to @hhhjort for PBS-Go implementation.

**Comment by SyntaxNode** [comment]
@bretg I'm not familiar with the `pri[0]` style of syntax for a yaml array. I expect this will more like this:
```
cookie-sync:
    coop-sync:
        default: true
        pri:
            - [bidderA, bidderB]
            - [bidderC, bidderD]
```

Is this how it was implemented in PBS-Java?

**Comment by SyntaxNode** [comment]
Implemented in PBS-Go 0.173.0.
